gse-afs-ext-base/src/main/java/com/powsybl/gse/afs/ext/base/VirtualScriptCreator.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.gse.afs.ext.base;
+
+import com.powsybl.afs.Project;
+import com.powsybl.afs.ext.base.AbstractScript;
+import com.powsybl.afs.ext.base.ModificationScript;
+import com.powsybl.gse.spi.GseContext;
+import com.powsybl.gse.util.NameTextField;
+import com.powsybl.gse.util.ProjectNodeSelectionPane;
+import javafx.application.Platform;
+import javafx.beans.binding.BooleanBinding;
+import javafx.scene.Scene;
+import javafx.scene.layout.ColumnConstraints;
+import javafx.scene.layout.GridPane;
+import javafx.scene.layout.Priority;
+
+import java.util.ResourceBundle;
+
+/**
+ * @author Nassirou Nambiema <nassirou.nambiena at rte-france.com>
+ */
+public class VirtualScriptCreator extends GridPane {
+
+    protected static final ResourceBundle RESOURCE_BUNDLE = ResourceBundle.getBundle("lang.VirtualScriptCreator");
+
+    private final NameTextField nameTextField;
+
+    private final ProjectNodeSelectionPane<ModificationScript> scriptSelectionPane;
+
+    public VirtualScriptCreator(AbstractScript script, Scene scene, GseContext context) {
+        Project project = script.getProject();
+        nameTextField = NameTextField.create(script);
+        scriptSelectionPane = new ProjectNodeSelectionPane<>(project, RESOURCE_BUNDLE.getString("ConcatenateScript"), scene.getWindow(), context, ModificationScript.class);
```

Comment:
you should prevent the user to select the current script as an inclusion.
or this would potentially cause an infinite loop when reading the total script.

Reply:
resolved